packages/gatsby/lib/utils/query-runner.js

```diff
@@ -221,96 +222,69 @@ const writeChildRoutes = () => {
 }
 const debouncedWriteChildRoutes = _.debounce(writeChildRoutes, 250)
 
-const babelPlugin = function({ types: t }) {
-  return {
-    visitor: {
-      TemplateLiteral(path, state) {
-        if (
-          path.parentPath.parentPath.parentPath.type !==
-          `ExportNamedDeclaration`
-        ) {
-          return
-        }
-        const exportPath = path.parentPath.parentPath.parentPath
-        const name = _.get(
-          exportPath,
-          `node.declaration.declarations[0].id.name`
-        )
-        if (name === `pageQuery`) {
-          const quasis = _.get(path, `node.quasis`, [])
-          const expressions = path.get(`expressions`)
-          const chunks = []
-          quasis.forEach(quasi => {
-            chunks.push(quasi.value.cooked)
-            const expr = expressions.shift()
-            if (expr) {
-              chunks.push(
-                expr.scope.bindings[expr.node.name].path.get(
-                  `value`
-                ).parentPath.node.init.quasis[0].value.cooked
-              )
-            }
-          })
-          const query = chunks.join(``)
-          console.time(`graphql query time`)
-          const graphql = state.opts.graphql
-          //path.parentPath.replaceWithSourceString(`require('fixme.json')`);
-        }
-      },
-    },
-  }
-}
-
 // Queue for processing files
 const q = queue(
-  ({ file, graphql, directory }, callback) => {
-    const fileStr = fs.readFileSync(file, `utf-8`)
+  async ({ file, graphql, directory }, callback) => {
+    let fileStr = fs.readFileSync(file, `utf-8`)
     let ast
-    try {
-      ast = babylon.parse(fileStr, {
-        sourceType: `module`,
-        sourceFilename: true,
-        plugins: [`*`],
-      })
-    } catch (e) {
-      console.log(`Failed to parse ${file}`)
-      console.log(e)
+    // Preprocess and attempt to parse source; return an AST if we can, log an error if we can't.
+    // I'm unconvinced that this is an especially good implementation...
```

Comment:
Haha, I typically assume my implementations are bad unless proven otherwise. I couldn't think of a better way to judge which plugin "wins".